auth/api/iam/jar.go

```diff
@@ -0,0 +1,172 @@
+package iam
+
+import (
+	"context"
+	"crypto"
+	"errors"
+	"net/url"
+
+	"github.com/lestrrat-go/jwx/v2/jwt"
+	"github.com/nuts-foundation/go-did/did"
+	"github.com/nuts-foundation/nuts-node/auth"
+	"github.com/nuts-foundation/nuts-node/auth/oauth"
+	cryptoNuts "github.com/nuts-foundation/nuts-node/crypto"
+	"github.com/nuts-foundation/nuts-node/vdr/resolver"
+)
+
+// requestObjectModifier is a function that modifies the Claims/params of an unsigned or signed (JWT) OAuth2 request
+type requestObjectModifier func(claims map[string]string)
+
+type requestObject struct {
+	Claims oauthParameters `json:"claims"`
+	Token  string          `json:"token,omitempty"`
+}
+
+func (ro requestObject) client() did.DID {
+	// set by JAR.Create, so always present
+	return did.MustParseDID(ro.Claims.get(oauth.ClientIDParam))
+}
+
+func (ro requestObject) nonce() string {
+	// set by JAR.Create, so always present
+	return ro.Claims.get(oauth.NonceParam)
+}
+
+func (ro requestObject) signed() bool {
+	return ro.Token != ""
+}
+
+var _ JAR = &jar{}
+
+type jar struct {
+	auth        auth.AuthenticationServices
+	jwtSigner   cryptoNuts.JWTSigner
+	keyResolver resolver.KeyResolver
+}
+
+type JAR interface {
```

Comment:
Maybe split it in JARCreater and JARParser? Since I'd expected `JAR` to be the signed request object itself

Reply:
Does this follow the design pattern Wout used in his DPoP PR (can it?)? Since it's sort of the same thing/principle

Reply:
I've left this as is since the goal is to logically group some methods together so they can be mocked. Might change when the `request_uri_method=post` is added. (next PR)

As for comparison to the `DPoP` PR. I see the resemblance between the two, but I don't want to mirror it since that requires moving `JAR` to the `crypto` module. The `JWTSigner` interface currently meets the needs we had for `did:nuts`, but not for `did:web` I think that interface should be updated so we don't pollute the `crypto` package.